vcd/resource_vcd_org_ldap.go

```diff
@@ -0,0 +1,420 @@
+			},
+			"telephone": { // Telephone
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "LDAP attribute to use for the user's telephone number. For example, telephoneNumber",
+			},
+			"group_membership_identifier": { // GroupMembershipIdentifier
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "LDAP attribute that identifies a user as a member of a group. For example, dn",
+			},
+			"group_back_link_identifier": { // GroupBackLinkIdentifier
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "LDAP attribute that returns the identifiers of all the groups of which the user is a member",
+			},
+		},
+	},
+}
+
+// resourceLdapGroupAttributes defines the elements of types.OrgLdapGroupAttributes
+// The field names are the ones used in the GUI, with a comment to indicate which API field each one corresponds to
+var resourceLdapGroupAttributes = &schema.Schema{
+	Type:        schema.TypeList,
+	Required:    true,
+	MaxItems:    1,
+	Description: "Group settings when `ldap_mode` is CUSTOM",
+	Elem: &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"object_class": { // ObjectClass
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "LDAP objectClass of which imported groups are members. For example, group",
+			},
+			"unique_identifier": { // ObjectIdentifier
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "LDAP attribute to use as the unique identifier for a group. For example, objectGuid",
+			},
+			"name": { // GroupName
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "LDAP attribute to use for the group name. For example, cn",
+			},
+			"membership": { // Membership
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "LDAP attribute to use when getting the members of a group. For example, member",
+			},
+			"group_membership_identifier": { // MembershipIdentifier
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "LDAP attribute that identifies a group as a member of another group. For example, dn",
+			},
+			"group_back_link_identifier": { // BackLinkIdentifier
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "LDAP group attribute used to identify a group member",
+			},
+		},
+	},
+}
+
+// resourceVcdOrgLdap defines types.OrgLdapSettingsType
+// The field names are the ones used in the GUI, with a comment to indicate which API field each one corresponds to
+func resourceVcdOrgLdap() *schema.Resource {
+	return &schema.Resource{
+		ReadContext:   resourceVcdOrgLdapRead,
+		CreateContext: resourceVcdOrgLdapCreate,
+		UpdateContext: resourceVcdOrgLdapUpdate,
+		DeleteContext: resourceVcdOrgLdapDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceVcdOrgLdapImport,
+		},
+		Schema: map[string]*schema.Schema{
+			"org_name": {
```

Comment:
I have a doubt/suggestion here: What about using the Organization ID instead? Taking advantage of the `vcd_org` data source, like this:

```hcl
data "vcd_org" "my-org" {
  name = "my-org"
}

resource "vcd_org_ldap" "my-ldap" {
  org_id = data.vcd_org.my-org.id
}
```
Wouldn't we benefit from the implicit dependency that Terraform manages automatically?

Reply:
OK. I submit to the majority. Changing it to `org_id`.